### PR TITLE
Shape Propagation Pass: Fix AdaptiveAveragePooling2d 

### PR DESCRIPTION
**Pull request by priyaramani**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #63629

Differential Revision: [D30461727](https://our.internmc.facebook.com/intern/diff/D30461727)

### CODE REVIEW

**Comment by priyaramani** [comment]
What? 
Bug fix for adaptive_average_pooling2d operator in shape propagation pass.

How? 
Implement checks and shape computation corresponding to the actual implementation of the operator

**Review by eellison — APPROVED** [review]
🚢  🚢  🚢

File: torch/csrc/jit/runtime/symbolic_shape_registry.cpp [hunk 1]
```diff
@@ -36,10 +36,17 @@ const std::string shape_compute_functions =
...
-          # TODO: return out directly, list len refiner would need to
-          # annotate the List Type with len directly in IR
           assert len(out) == 2
-          return [out[0], out[1]]
+          assert len(self) == 3 or len(self) == 4
+          for i in range (1, len(self)):
+            assert self[i] != 0
+
+          shape: List[int] = []
+          for i in range(0, len(self)-2):
```
[the remark this answers is not in the document]
Comment (priyaramani) [reply torch/csrc/jit/runtime/symbolic_shape_registry.cpp]: Thanks, updated!

**Comment by priyaramani** [comment]
@priyaramani has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30461727).

**Comment by priyaramani** [comment]
@priyaramani has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30461727).

**Comment by priyaramani** [comment]
@priyaramani has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30461727).

**Comment by priyaramani** [comment]
@priyaramani has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30461727).

**Pull request closed without merging**